ts/src/base/functions/number.ts

```diff
@@ -39,6 +39,7 @@ const precisionConstants = {
 
 function numberToString (x) { // avoids scientific notation for too large and too small numbers
     if (x === undefined) return undefined;
+    if (typeof x === 'string') return x;
```

Comment:
This seems unnecessary and actually less efficient, seeing as it does a string comparison https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toString#description

> The String object overrides the toString method of Object; it does not inherit Object.prototype.toString(). For String values, the toString method returns the string itself (if it's a primitive) or the string that the String object wraps. It has the exact same implementation as String.prototype.valueOf().

Reply:
I am not sure it's less efficient, you can test out here: https://jsbench.me/91lvmdl2e3/1
also tried other testers, everywhere this approach wins

Reply:
I don't understand why this approach would be a gain, because the toString method of a string just returns the string itself, so having another conditional just seems like extra, but this is such a minor thing that it doesn't really matter